src/client/LiveConf.js

```diff
@@ -160,56 +161,172 @@ class LiveConf {
     }
   }
 
-  calcSegDuration({sourceTimescale}) {
+  /*
+   * Calculates mez segment durations based on input stream parameters
+   *
+   * Live input formats have fixed timebase:
+   * - MPEG-TS/SRT input stream timebase is 90000
+   * - RTMP input stream timebase is 1000 and gets translated to 16000 if not otherwise specified
+   *
+   * This causes frame duration irregularities for certain frame rates.
+   * For example RTMP 60fps has frames of durations 16 and 17.  MPEG-TS 59.94fps has frames of
+   * durations 1001 and 1002.
+   *
+   * Live mez segmentation requires that the segment be cut at the specific number of frames, and when
+   * the frame durations are irregular we adjust both the video timebase and the video frame duration
+   * to make the math possible.  This adjustment is also required for live-to-vod conversion.
+   *
+   * For example for MPEG-TS 59.94fps, the mez segment timebase needs to be 60000
+   * (and resulting frame duration is 1001) and for RTMP 60fps the timebase needs to be 15360 (resulting frame
+   * duration is 256).
+   *
+   * @sourceTimescale - adjusted source video stream timescale (eg. MPEGTS 90000, RTMP 16000 )
+   * @sampleRate - audio sample rate (commonly 48000 but can be different)
+   * @audioCodec - audio codec as a string (eg. "aac")
+   * @return - segment encoding parameters
+   */
+  calcSegDuration({sourceTimescale, sampleRate, audioCodec}) {
+    let seg = {};
+
+    switch(this.probeKind()) {
+      case "rtmp":
+        seg = this.calcSegDurationRtmp({sourceTimescale, sampleRate, audioCodec});
+        break;
+      case "udp":
+      case "srt":
+        seg = this.calcSegDurationMpegts({sourceTimescale, sampleRate, audioCodec});
+        break;
+      default:
+        throw "protocol not supported - " + this.probeKind();
+    }
+
+    if(audioCodec == "aac") {
+      seg.audio = 29.76 * sampleRate;
```

Comment:
I don't know if the math here is ever working with large enough numbers for it to matter re. floating point inaccuracy, but it might be smart to use a specialized library to ensure exact precision since the math here must be correct.

The video editor uses fraction.js for this

https://github.com/eluv-io/elv-video-editor/blob/master/src/utils/FrameAccurateVideo.js

https://github.com/rawify/Fraction.js

Reply:
Will be resolved in a separate branch: https://github.com/eluv-io/elv-client-js/tree/liveconf-timebase